apps/frontend/src/pages/project/project.tsx

```diff
@@ -1,43 +1,77 @@
 import {
 	Breadcrumbs,
 	Button,
+	Modal,
 	PageLayout,
 } from "~/libs/components/components.js";
 import { AppRoute, DataStatus } from "~/libs/enums/enums.js";
 import {
 	useAppDispatch,
 	useAppSelector,
+	useCallback,
 	useEffect,
 	useModal,
 	useParams,
 } from "~/libs/hooks/hooks.js";
-import { actions as projectActions } from "~/modules/projects/projects.js";
+import {
+	actions as projectActions,
+	type ProjectPatchRequestDto,
+} from "~/modules/projects/projects.js";
 import { NotFound } from "~/pages/not-found/not-found.jsx";
+import { ProjectUpdateForm } from "~/pages/projects/libs/components/components.js";
 
-import { SetupAnalyticsModal } from "./libs/components/components.js";
+import {
+	ProjectMenu,
+	SetupAnalyticsModal,
+} from "./libs/components/components.js";
 import styles from "./styles.module.css";
 
 const Project = (): JSX.Element => {
+	const handleManageAccess = useCallback(() => {}, []);
 	const dispatch = useAppDispatch();
 	const { id: projectId } = useParams<{ id: string }>();
 
-	const { project, projectStatus } = useAppSelector(({ projects }) => projects);
+	const { project, projectPatchStatus, projectStatus } = useAppSelector(
+		({ projects }) => projects,
+	);
 
 	const {
 		isOpened: isSetupAnalyticsModalOpened,
 		onClose: onSetupAnalyticsModalClose,
 		onOpen: onSetupAnalyticsModalOpen,
 	} = useModal();
 
+	const {
+		isOpened: isEditModalOpen,
+		onClose: handleEditModalClose,
+		onOpen: handleEditModalOpen,
+	} = useModal();
+
 	useEffect(() => {
 		if (projectId) {
 			void dispatch(projectActions.getById({ id: projectId }));
+
+			if (projectPatchStatus === DataStatus.FULFILLED) {
+				handleEditModalClose();
+			}
```

Comment:
Why not put it in a separate `useEffect`?

Reply:
I split them into two useEffects to keep the logic cleaner and easier to follow. One handles fetching the project, and the other manages the modal state.